### ISSUE TITLE
Question: does it work with proxy_next_upstream?

### ISSUE THREAD

**Issue by ckornacker** [jkeys089/lua-resty-resolver#19]
I was wondering whether [proxy_next_upstream](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_next_upstream) works when using the resolver if there are multiple A records for a single domain?

**Comment by jkeys089** [comment]
@ckornacker yes, it works. You can have as many `server` declarations as you like in the `upstream` block -- we'll just ignore them but I believe they are needed to enable `proxy_next_upstream`. However, we're just using simple round-robin load balancing when there are multiple A records so it is possible to get the same backend again when retrying. There is an [open feature request](https://github.com/jkeys089/lua-resty-resolver/issues/3) to add a backend health check but no one has started work on that as far as I know.

**Comment by ckornacker** [comment]
Thanks @jkeys089, that does seem to work to some extent. There are still occasions where I'm getting the same backend for all dummy `server` declarations (currently 5).

**Comment by jkeys089** [comment]
@ckornacker if you can provide a reproducer I'd be happy to take a look.
